variables.tf

```diff
@@ -93,6 +93,20 @@ variable "api_server_authorized_ip_ranges" {
   default     = null
 }
 
+variable "automatic_channel_upgrade" {
+  type        = string
+  default     = null
+  description = "(Optional) The upgrade channel for this Kubernetes Cluster. Possible values are  `patch`, `rapid`, `node-image` and `stable`. By default automatic-upgrades are turned off. See [the documentation](https://learn.microsoft.com/en-us/azure/aks/auto-upgrade-cluste) for more information"
+  validation {
```

Comment:
You should use the function `contains` here:

```
validation {
    condition = (
      var.automatic_channel_upgrade == null || (
        contains(["patch", "stable", "rapid", "node-image"], var.automatic_channel_upgrade)
      )
    )
    error_message = "`automatic_channel_upgrade`'s possible values are `patch`, `stable`, `rapid` or `node-image`."
  }
```

Reply:
Commenting on the tests have done earlier, would it work to validate like this ?
```
validation {
  condition = (
    var.automatic_channel_upgrade == null || (
      contains(["patch", "stable", "rapid", "node-image"], var.automatic_channel_upgrade) &&
      matches("^(?:[0-9]\\.){2}[0-9]$", var.kubernetes_version) == false
    )
  )
  error_message = "`automatic_channel_upgrade`'s possible values are `patch`, `stable`, `rapid` or `node-image` and `kubernetes_version` must not include a patch version number."
}
```

Reply:
Should the node-image channel be independent from the Kubernetes patch version check ?

Reply:
Hey sorry I totally missed that you cannot validate a variable referencing another variable value. But you could move this logic to the `azurerm_kubernetes_cluster` resource to validate at runtime.

Reply:
@zioproto I've used `contains` in the variable validation (to get fast feedback if the value is valid) and added a second more extensive precondition to the `azurerm_kubernetes_cluster` resource. Could you have a look at it?

Reply:
Resolving this for now since it's outdated with the changes suggested by @lonegunmanb.